### PR TITLE
Cleaned up generated index.js and package.js files

### PR DESCRIPTION
**Pull request by lourd**
The generated package.js and index.js files for the npm-container package look
much cleaner now. The ending comment with the line number is removed and
am using js-beautify to make it look pretty.

Updated
* switched from call `Package._transitional_registerBuildPlugin` to the current API of
`Package.registerBuildPlugin`

Removed:
* package.json from being added to npm-container package. It wasn't actually
being used, the package is reading from the packages.json file in the main app.
I removed it for clarity
* deleting packages/npm directory. I imagine this was providing some sort of
backwards compatibility, but when woring on the package locally it would delete
the directory right from under you while you're working on it. No longer!
* execSync function, it wasn't being used
* rimraf dependency, also wasn't being used

### CODE REVIEW

File: plugin/init_npm.js [hunk 1]
```diff
@@ -113,6 +97,6 @@ function _packageJsContent () {
   }
 
   Package.onUse(function(api) {
-    api.add_files(['index.js', '../../packages.json'], 'server');
```
Comment (arunoda) [inline plugin/init_npm.js]: This is needed. Then meteor watches packages.json file the root of your app and reload when it changes.

**Comment by lourd** [comment]
Ah, ok that's what that was doing. I amended the commit to add it back in, add a comment, and update the commit message.

**Comment by lourd** [comment]
I also just removed the `process.exit` call becase Meteor will automatically restart the build on the addition of npm-container to the `.meteor/packages` file

**Comment by arunoda** [comment]
I had to add process.exit(0).
See comments: https://github.com/meteorhacks/npm/commit/3f27c38f59b1ab98a4964cec146480efe0ada08d

**Comment by lourd** [comment]
Interesting, I thought I'd tested what you described in the comment. I made a fresh meteor project, added my forked `npm` package that didn't have the `process.exit` call, defined a `package.json`, and then started the project with `meteor`. It brought in the npm dependencies without a problem — I could see them in the build and call them on the server in the shell. Then I did a `meteor reset` and then `meteor build` and it brought in the dependencies just fine into the built project.

What's the case where this doesn't occur?

**Comment by arunoda** [comment]
Okay do this.

* there's only `packages.json` in the repo
* no npm-container in the .meteor/packages
* no packages/npm-container

Then simply invoke `meteor`

> This is a pattern used by some developers.

**Comment by lourd** [comment]
Right, those are the steps I was trying to describe. And Meteor worked fine without `process.exit` being invoked. It appears that when `npm-container` gets added to `.meteor/packages` it triggers the build to restart.

Then I took it one step further. With the same conditions you described, invoke `meteor build`. Same positive results — the build restarts when the `npm-container` Meteor package is created and all dependencies declared in `packages.json` are brought into the bundle accordingly.

--

**Pull request merged**